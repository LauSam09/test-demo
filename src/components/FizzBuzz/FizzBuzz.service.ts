export function fizzOrBuzz(number: number): string | number {
  if (number % 15 === 0) {
    return `${number} - FizzBuzz!`;
  } else if (number % 3 === 0) {
    return `${number} - Fizz!`;
  } else if (number % 5 === 0) {
    return `${number} - Buzz!`;
  } else {
    return number;
  }
}
