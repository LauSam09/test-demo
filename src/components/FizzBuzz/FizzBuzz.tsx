import React, { useState } from "react";

import { fizzOrBuzz } from "./FizzBuzz.service";

export default function FizzBuzz() {
  const [number, setNumber] = useState(0);

  return (
    <>
      <span>{fizzOrBuzz(number)}</span>
      <button onClick={() => setNumber((n) => n + 1)}>Click me</button>
    </>
  );
}
