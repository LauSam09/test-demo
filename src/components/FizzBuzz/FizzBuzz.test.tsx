import React from "react";
import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";

import FizzBuzz from "./FizzBuzz";
import { fizzOrBuzz } from "./FizzBuzz.service";

jest.mock("./FizzBuzz.service");

describe("<FizzBuzz />", () => {
  test("should match snapshot", () => {
    // Act
    const tree = renderer.create(<FizzBuzz />);

    // Assert
    expect(tree).toMatchSnapshot();
  });

  test("should increment number when button is clicked", () => {
    // Arrange
    (fizzOrBuzz as jest.Mock).mockImplementation((number: number) => number);
    const { getByText, queryByText } = render(<FizzBuzz />);
    const button = getByText("Click me");

    // Assert
    expect(queryByText(0)).toBeInTheDocument();

    // Act
    fireEvent.click(button);

    // Assert
    expect(queryByText(1)).toBeInTheDocument();
  });
});
