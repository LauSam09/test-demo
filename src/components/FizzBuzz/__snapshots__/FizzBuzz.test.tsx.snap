// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FizzBuzz /> should match snapshot 1`] = `
Array [
  <span>
    0
  </span>,
  <button
    onClick={[Function]}
  >
    Click me
  </button>,
]
`;
